src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,39 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
 public class SalaryInfo {
+    private static final DateTimeFormatter DATE_TIME_FORMATTER = DateTimeFormatter
+            .ofPattern("dd.MM.yyyy");
+
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        LocalDate fromDate = LocalDate.parse(dateFrom, DATE_TIME_FORMATTER);
+        LocalDate toDate = LocalDate.parse(dateTo, DATE_TIME_FORMATTER);
+
+        StringBuilder result = new StringBuilder();
+        result.append("Report for period ").append(dateFrom)
+                .append(" - ").append(dateTo);
+
+        for (String name : names) {
+            int totalSalary = 0;
+            for (String record : data) {
+                String[] recordParts = record.split(" ");
+                LocalDate date = LocalDate.parse(recordParts[0], DATE_TIME_FORMATTER);
+                if (date.isBefore(fromDate) || date.isAfter(toDate)) {
+                    continue;
+                }
+                String nameAtRecord = recordParts[1];
+                if (name.equals(nameAtRecord)) {
+                    int hoursAtRecord = Integer.parseInt(recordParts[2]);
+                    int moneyPerHour = Integer.parseInt(recordParts[3]);
```

Comment:
for indexes
![image](https://github.com/mate-academy/jv-salary-info/assets/20326858/e23b69c3-6ed2-43a2-841e-38bff57e9b1e)